src/som/interpreter/actors/EventualMessage.java

```diff
@@ -323,24 +331,40 @@ public static EventualMessage getCurrentExecutingMessage() {
 
   /**
    * Indicates that execution should stop and yield to the debugger,
-   * before the computed value is used to resolve the promise.
+   * before the message is processed.
+   */
+  public boolean haltOnReceive() {
+    return haltOnReceive;
```

Comment:
You haven't answered my question. So, why is this confusing?
And which conventions. Do you have a specific link I should be looking at?
I am not a Java native.